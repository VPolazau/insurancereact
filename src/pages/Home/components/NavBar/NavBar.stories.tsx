import { Story, Meta } from '@storybook/react';

import { NavBar } from './NavBar';

export default {
    title: 'UI/NavBar',
    component: NavBar,
} as Meta;

const Template: Story = (args) => <NavBar {...args} />;

export const Default = Template.bind({});
Default.args = {};
