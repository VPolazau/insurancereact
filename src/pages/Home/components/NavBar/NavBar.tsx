import React from 'react';
import { useNavigate } from 'react-router-dom';

import { DropDownList } from '../DropDownList';
import { NAVIGATION_DROP_LINKS } from '../../../../utils/constants';

import styles from './styles.module.css';

export const NavBar: React.FC = () => {
    const navigate = useNavigate();

    const handleSpecialButtonClick = () => {
        navigate('/account/policies/insurance-case');
    };

    return (
        <nav className={styles.navbar} data-testid="navbarComponent">
            <div className={styles.menu}>
                {NAVIGATION_DROP_LINKS.map(({ mainLinkPath, mainLinkValue, dropNavLinks }) => (
                    <DropDownList
                        key={mainLinkValue}
                        mainLinkPath={mainLinkPath}
                        mainLinkValue={mainLinkValue}
                        dropNavLinks={dropNavLinks}
                    />
                ))}
            </div>

            <button className={styles.buttonSpecial} onClick={handleSpecialButtonClick}>
                Страховой случай
            </button>
        </nav>
    );
};
