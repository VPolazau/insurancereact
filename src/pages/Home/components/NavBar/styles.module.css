.navbar {
  height: 34px;
  display: flex;
  margin: 20px auto;
  padding: 0 24px;
  /* ширина по макету + 2 падинга */
  max-width: calc(1218px + 48px);
  width: 100%;
  justify-content: space-between;
  align-items: center;
}

.menu {
  display: flex;
  align-items: center;
  gap: 52px;
}

.buttonSpecial {
  font-family: Nunito Sans, sans-serif;
  font-size: 17px;
  font-weight: 400;
  line-height: 22px;
  letter-spacing: 0px;
  color: #1F1D1D;  
  font-style: normal;
  display: flex;
  align-items: center;
}

.buttonSpecial::before {
  content: '';
  display: inline-block;
  width: 24px;
  height: 24px;
  margin-right: 13px;
  background: url(../../../../assets/img/lightning.svg) no-repeat;
  background-size: contain;
  transform: rotate(180deg);
}

@media (max-width: 650px) {
  .navbar {
    display: none;
  }
}