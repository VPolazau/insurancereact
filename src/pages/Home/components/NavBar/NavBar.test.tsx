import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { NavBar } from './NavBar';

describe('NavBar Component', () => {
    it('button-1', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(screen.getByText('Авто')).toBeInTheDocument();
    });
    it('button-2', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(screen.getByText('Имущество')).toBeInTheDocument();
    });
    it('button-3', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(screen.getByText('Здоровье')).toBeInTheDocument();
    });
    it('button-4', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(screen.getByText('Отдых')).toBeInTheDocument();
    });
    it('should render the special button', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(screen.getByText('Страховой случай')).toBeInTheDocument();
    });
    it('should match the snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
