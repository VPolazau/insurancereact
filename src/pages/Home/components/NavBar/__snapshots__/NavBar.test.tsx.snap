// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should match the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    data-testid="navbarComponent"
  >
    <div
      class="menu"
    >
      <div
        class="dropDownListBasic"
        data-testid="dropDownList"
      >
        <a
          class="dropDownMainLink"
          href="/in-progress"
        >
          Авто
        </a>
        <div
          class="dropDownContent"
        >
          <a
            class="dropDownText"
            href="/in-progress"
          >
            КАСКО
          </a>
          <a
            class="dropDownText"
            href="/in-progress"
          >
            ОСАГО
          </a>
        </div>
      </div>
      <div
        class="dropDownListBasic"
        data-testid="dropDownList"
      >
        <a
          class="dropDownMainLink"
          href="/in-progress"
        >
          Имущество
        </a>
        <div
          class="dropDownContent"
        >
          <a
            class="dropDownText"
            href="/property/apartment"
          >
            Квартира
          </a>
          <a
            class="dropDownText"
            href="/in-progress"
          >
            Дом
          </a>
        </div>
      </div>
      <div
        class="dropDownListBasic"
        data-testid="dropDownList"
      >
        <a
          class="dropDownMainLink"
          href="/in-progress"
        >
          Здоровье
        </a>
        <div
          class="dropDownContent"
        >
          <a
            class="dropDownText"
            href="/in-progress"
          >
            ДМС
          </a>
          <a
            class="dropDownText"
            href="/in-progress"
          >
            Страхование жизни
          </a>
        </div>
      </div>
      <div
        class="dropDownListBasic"
        data-testid="dropDownList"
      >
        <a
          class="dropDownMainLink"
          href="/in-progress"
        >
          Отдых
        </a>
        <div
          class="dropDownContent"
        >
          <a
            class="dropDownText"
            href="/in-progress"
          >
            Страхование путешественников
          </a>
        </div>
      </div>
    </div>
    <button
      class="buttonSpecial"
    >
      Страховой случай
    </button>
  </nav>
</DocumentFragment>
`;
